### PR TITLE
Topological modalities

### PR DESCRIPTION
**Pull request by awswan**
I needed some results about topological modalities for a project, and I think it makes sense to add these to the library in case anyone else wants to use them. I'm not 100% sure about the naming, so feel free to change if necessary.  There are a couple of naming clashes that come from the literature, so there's not  much I can do about them: 1. Separated is a general definition that applies to any modality, whereas in the library (and many places in the literature) it is specific to the double negation modality; 2. So far Injective in the library is only referring to functions, I think, but I also use it for injective objects, as in https://ncatlab.org/nlab/show/injective+object .

### CODE REVIEW

File: Cubical/HITs/Nullification/Topological.agda [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  equivPreservesIsNull (invEquiv e) (isNullEquiv (snd X) (snd Y))
+  where
+    e : (X ≡ Y) ≃ ((fst X) ≃ (fst Y))
+    e =
+      X ≡ Y ≃⟨ invEquiv (Σ≡PropEquiv λ _ → isPropIsNull) ⟩
+      (fst X) ≡ (fst Y) ≃⟨ univalence ⟩
+      (fst X) ≃ (fst Y) ■
+
+module _ (isPropS : (α : A) → isProp (S α)) where
+  {- Recall that a type Z is injective when can extend any map S α → Z to
```
Comment (felixwellen) [inline Cubical/HITs/Nullification/Topological.agda:57]: Typo, "when we can extend"

File: Cubical/HITs/Nullification/Topological.agda [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  ℓ ℓ' : Level
+
+{-
+  We use the formulation of lexness that the universe of null types is
+  null itself.
+
+  We choose the level of the universe of null types to be as low as
+  possible such that it is null when the generators are propositions.
+-}
+nullTypes : Type (ℓ-max (ℓ-suc (ℓ-max ℓ ℓs)) ℓα )
```
Comment (felixwellen) [inline Cubical/HITs/Nullification/Topological.agda:40]: "NullType" would be more in line with the naming conventions.

**Review by felixwellen — CHANGES_REQUESTED** [review]
Just a typo and a renaming suggestion, otherwise ready to merge. Thanks for contributing!

**Comment by awswan** [comment]
Thanks! I've made both those changes.

**Pull request merged**